storage/config.go

```diff
@@ -42,48 +44,50 @@ func (s *Storage) SetDelegate(sk, pkh string) error {
 }
 
 func (s *Storage) GetSignerType() (int, error) {
-	var st int = 0
+
+	var signerType int = 0
 
 	err := s.db.View(func(tx *bolt.Tx) error {
-		b := tx.Bucket([]byte(CONFIG_BUCKET))
-		_st := b.Get([]byte(SIGNER_TYPE))
-		if _st != nil {
-			st = btoi(_st)
+		bucket := tx.Bucket([]byte(CONFIG_BUCKET))
+		signerTypeBytes := bucket.Get([]byte(SIGNER_TYPE))
+		if signerTypeBytes != nil {
+			signerType = int(btoi(signerTypeBytes))
 		}
 		return nil
 	})
 
-	return st, err
+	return signerType, err
 }
 
-func (s *Storage) SetSignerType(d int) error {
+func (s *Storage) SetSignerType(signerType int) error {
 	return s.db.Update(func(tx *bolt.Tx) error {
-		b := tx.Bucket([]byte(CONFIG_BUCKET))
-		return b.Put([]byte(SIGNER_TYPE), itob(d))
+		bucket := tx.Bucket([]byte(CONFIG_BUCKET))
+		return bucket.Put([]byte(SIGNER_TYPE), itob(int(signerType)))
 	})
 }
 
-func (s *Storage) GetSignerSk() (string, error) {
-	var sk string
-
-	err := s.db.View(func(tx *bolt.Tx) error {
-		b := tx.Bucket([]byte(CONFIG_BUCKET))
-		sk = string(b.Get([]byte(SIGNER_SK)))
-		return nil
+func (s *Storage) GetSignerSk() (signedSK string, err error) {
```

Comment:
The boltdb docs say that Get() values are only valid during the transaction and that you have to either copy() or save to previously declared var. I'm not sure if this change works.